src/app.py

```diff
@@ -335,11 +335,28 @@ def query(
                 rule_author=rule.author,
                 is_global=rule.is_global,
                 is_private=rule.is_private,
+                is_degenerate=rule.parse().is_degenerate,
                 parsed=rule.parse().query,
             )
             for rule in rules
         ]
 
+    degenerate_rules = [r.name for r in rules if r.parse().is_degenerate]
+    disallow_degenerate = db.get_mquery_config_key("query_disallow_degenerate")
+    if degenerate_rules and (disallow_degenerate == "true"):
```

Comment:
Seems like `disallow_degenerate == "true"` should be somewhere lower in the abstraction layer :thinking: (just a thought)

Reply:
Also, I don't think the parentheses are needed?
```suggestion
    if degenerate_rules and disallow_degenerate == "true":
```

Reply:
O, and of course you're right about the lower abstraction layer. It's planned and scheduled for the next release: #257.

The current config is a mess anyway, for example I've named it "disallow_degenerate" instead of "allow_degenerate", because empty value is the default, and I want the default to be allowed. Letting storage details leak into config key names is not the best abstraction.